wicket-core/src/main/java/org/apache/wicket/page/PageAccessSynchronizer.java

```diff
@@ -93,8 +94,8 @@ public Duration getTimeout(int pageId)
 	 */
 	public void lockPage(int pageId) throws CouldNotLockPageException
 	{
-		final Thread thread = Thread.currentThread();
-		final PageLock lock = new PageLock(pageId, thread);
+		final RequestCycle cycle = RequestCycle.get();

```

Comment:
Sure, we could keep the thread reference. But when the request is suspended, a certain time *no* thread is keeping the lock (the page is still locked though) and from a certain point in time another thread would pick up the lock.